src/map.c

```diff
@@ -3,752 +3,625 @@
  * "LICENSE" for information on usage and redistribution of this file.
  */
 
-#include <assert.h>
-#include <stdint.h>
-#include <stdlib.h>
-#include <string.h>
-
-#include "map.h"
-
-struct map_internal {
-    struct map_node *head;
-
-    /* Properties */
-    size_t key_size, element_size, size;
-
-    map_iter_t it_end, it_most, it_least;
-
-    int (*comparator)(const void *, const void *);
-};
-
-typedef enum { RB_RED = 0, RB_BLACK } map_color_t;
-
 /*
- * Get parent of node
- * @node: pointer to the rb node
- *
- * Return: rb parent node of @node
+ * This map implementation has undergone extensive modifications, heavily
+ * relying on the rb.h header file from jemalloc.
+ * See
```

Comment:
It should be
```c
 * See https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/rb.h .
 ```

Reply:
clang-format will make it weird every round.
How can I bypass that?

Reply:
Try `/* clang-format off */` and `/* clang-format on */`.